pkg/upstream/cluster/cluster_manager.go

```diff
@@ -79,9 +80,19 @@ func (singleton *clusterManagerSingleton) Destroy() {
 	clusterManagerInstance.clusterManager = nil
 }
 
+type ClusterManagerOption func(types.ClusterManager)
+
+func WithClusterPoolEnable(clusterPoolEnable bool) ClusterManagerOption {
+	return func(cm types.ClusterManager) {
+		if c, ok := cm.(*clusterManagerSingleton); ok {
+			c.clusterPoolEnable = clusterPoolEnable
+		}
+	}
+}
+
 var clusterManagerInstance = &clusterManagerSingleton{}
 
-func NewClusterManagerSingleton(clusters []v2.Cluster, clusterMap map[string][]v2.Host, tls *v2.TLSConfig) types.ClusterManager {
+func NewClusterManagerSingleton(clusters []v2.Cluster, clusterMap map[string][]v2.Host, tls *v2.TLSConfig, opts ...ClusterManagerOption) types.ClusterManager {
```

Comment:
> We need to consider the scenario where this configuration is modified at running.

I can add an interface in `ClusterManager` , but it will be not compatible compared with last version.

Reply:
The old connpools will not be used anymore where this configuration is modified from disable to enable, and we should shutdown them. But this is more complex to do, and then there are very few scenarios to modify this at running, so you can leave it now and record a TODO